lib/preload.js

```diff
@@ -196,17 +198,26 @@ const getLayerSize = (registryHost, token, imageRepo, blobSum) => {
 }
 
 const getContainerSize = (apiHost, apiToken, registryHost, imageRepo) => {
-  console.log('Fetching container size for', imageRepo)
+  const progress = new visuals.Progress(`Fetching container size for ${imageRepo}`)
+  progress.update({ percentage: 0 })
   return getRegistryToken(apiHost, apiToken, registryHost, imageRepo)
   .then((token) => {
     return registry(registryHost, `v2/${imageRepo}/manifests/latest`, token, {}, true, true)
     .then((response) => {
       const layersCount = _.countBy(_.map(response.body.fsLayers, 'blobSum'))
+      const size = _.size(layersCount) + 1
+      let finished = 1
+      const updateProgress = () => {
+        progress.update({ percentage: finished / size * 100 })
+      }
+      updateProgress()
       return Promise.map(
         _.entries(layersCount),
         (layer) => {
           return getLayerSize(registryHost, token, imageRepo, layer[0])
           .then((layerSize) => {
+            finished += 1
+            updateProgress()
```

Comment:
Doesn't matter, but I guess you could have initialised `let finished = 0` and then done `finished += 1` as the first line inside `updateProgress` ?